Wikipedia/Code/WMFNotificationsController.m

```diff
@@ -0,0 +1,99 @@
+NSString *const WMFNotificationInfoArticleURLStringKey = @"articleURLString";
+NSString *const WMFNotificationInfoThumbnailURLStringKey = @"thumbnailURLString";
+NSString *const WMFNotificationInfoArticleExtractKey = @"articleExtract";
+NSString *const WMFNotificationInfoStoryHTMLKey = @"storyHTML";
+NSString *const WMFNotificationInfoViewCountsKey = @"viewCounts";
+
+@interface WMFNotificationsController ()
+@property (nonatomic, strong) dispatch_queue_t notificationQueue;
+@property (nonatomic, strong) dispatch_source_t notificationSource;
+@end
+
+@implementation WMFNotificationsController
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        self.notificationQueue = dispatch_queue_create("org.wikimedia.notifications", DISPATCH_QUEUE_SERIAL);
+    }
+    return self;
+}
+
+- (void)start {
+    [self requestAuthenticationIfNecessaryWithCompletionHandler:^(BOOL granted, NSError *_Nullable error) {
+        if (error) {
+            DDLogError(@"Error requesting authentication: %@", error);
+        }
+        //        dispatch_async(self.notificationQueue, ^{
+        //            self.notificationSource = dispatch_source_create(DISPATCH_SOURCE_TYPE_TIMER, 0, 0, self.notificationQueue);
+        //            dispatch_source_set_timer(self.notificationSource, DISPATCH_TIME_NOW, WMFNotificationUpdateInterval * NSEC_PER_SEC, WMFNotificationUpdateInterval * NSEC_PER_SEC / 10);
+        //            dispatch_source_set_event_handler(self.notificationSource, ^{
+        //                [self sendNotification];
+        //            });
+        //            dispatch_resume(self.notificationSource);
+        //        });
+    }];
+}
+
+- (void)requestAuthenticationIfNecessaryWithCompletionHandler:(void (^)(BOOL granted, NSError *__nullable error))completionHandler {
+
+    UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
+    UNNotificationAction *readNowAction = [UNNotificationAction actionWithIdentifier:WMFInTheNewsNotificationReadNowActionIdentifier title:NSLocalizedString(@"in-the-news-notification-read-now-action-title", nil) options:UNNotificationActionOptionForeground];
+    UNNotificationAction *saveForLaterAction = [UNNotificationAction actionWithIdentifier:WMFInTheNewsNotificationShareActionIdentifier title:NSLocalizedString(@"in-the-news-notification-share-action-title", nil) options:UNNotificationActionOptionForeground];
+    UNNotificationAction *shareAction = [UNNotificationAction actionWithIdentifier:WMFInTheNewsNotificationSaveForLaterActionIdentifier title:NSLocalizedString(@"in-the-news-notification-save-for-later-action-title", nil) options:UNNotificationActionOptionForeground];
+
+    UNNotificationCategory *inTheNewsCategory = [UNNotificationCategory categoryWithIdentifier:WMFInTheNewsNotificationCategoryIdentifier actions:@[readNowAction, saveForLaterAction, shareAction] intentIdentifiers:@[] options:UNNotificationCategoryOptionNone];
```

Comment:
@montehurd updated to nil check those items. Were you on a different locale? Did you see which was nil?